teamsupport/models.py

```diff
@@ -0,0 +1,57 @@
+from property_caching import cached_property
+
+from teamsupport.errors import MissingArgumentError
+
+
+class Ticket(object):
+    def __init__(self, client, ticket_id=None, data=None):
+        self.client = client
+        self._data = data
+        if ticket_id:
+            self._data = self.client.get_ticket(ticket_id)
+        elif not (ticket_id or data):
+            raise MissingArgumentError(
+                "__init__() needs either a 'ticket_id' or 'data' argument "
+                "(neither given)")
+        self.id = self.TicketID.text
+
+    def __getattr__(self, name):
+        if self.data.find(name) is not None:
+            return self.data.find(name)
```

Comment:
Would it make sense to `.text` here?

`self.TicketID.text` doesn't seem like a beautiful interface to me.